esp-hal/src/uart.rs

```diff
@@ -510,57 +426,94 @@ where
     }
 }
 
+impl<'d, T> UartRx<'d, T, Blocking>
+where
+    T: Instance + 'd,
+{
+    /// Create a new UART RX instance in [`Blocking`] mode.
+    ///
+    /// Use [crate::gpio::NO_PIN] for the `cts` parameter if CTS is not used.
+    pub fn new<RX: InputPin, CTS: InputPin>(
+        _uart: impl Peripheral<P = T> + 'd,
+        rx: impl Peripheral<P = RX> + 'd,
+        cts: Option<impl Peripheral<P = CTS> + 'd>,
```

Comment:
Let's keep things consistent by offering the `with_x` for cts/rts on the two sides of the uart